examples/usecase-passport/auth.ts

```diff
@@ -0,0 +1,134 @@
+import { randomBytes } from "crypto";
+import { Router } from "express";
+import { Passport, Profile } from "passport";
+import { Strategy, StrategyOptions } from "passport-github2";
+import { hashSync } from "bcryptjs";
+import { createAuth } from "@keystone-6/auth";
+import { statelessSessions } from "@keystone-6/core/session";
+import type { KeystoneContext } from "@keystone-6/core/types";
+import type { VerifyCallback, VerifyFunction } from "passport-oauth2";
+import type { Lists, TypeInfo } from ".keystone/types";
+import type { User as PrismaUser } from ".myprisma/client";
+
+type BaseSession<TData extends Record<string, unknown>> = {
+  listKey: string;
+  itemId: string;
+  // Usually is added by @keystone-6/auth if enabled
+  data?: TData;
+};
+
+type SessionData = {
+  name: string;
+  createdAt: string;
+};
+
+export type Session = BaseSession<SessionData>;
+
+export const { withAuth } = createAuth<Lists.User.TypeInfo<Session>>({
+  listKey: "User",
+  identityField: "email",
+  // Strictly typed sessionData (see Join at the end)
+  sessionData: "name createdAt" satisfies Join<keyof SessionData>,
```

Comment:
Strict typing won't work. I forgot this is a GQL query.

Note: Remove satisfies